### PR TITLE
MAT-5213 Nested libraries are not included in Measure bundle

### PR DESCRIPTION
**Pull request by adongare**
## MADiE FHIR SERVICE

Jira Ticket: [MAT-5213](https://jira.cms.gov/browse/MAT-5213)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/madiefhirservice/services/MeasureBundleService.java [hunk 1]
```diff
@@ -54,27 +58,33 @@ public List<Bundle.BundleEntryComponent> createBundleComponentsForLibrariesOfMad
...
 
+  public void getIncludedLibraries(String cql, List<Library> libraries) {
+    LibraryCqlVisitor visitor = libCqlVisitorFactory.visit(cql);
+    for (Pair<String, String> libraryNameValuePair : visitor.getIncludedLibraries()) {
+      Optional<Library> optionalLibrary =
+          hapiFhirServer.fetchHapiLibrary(
+              libraryNameValuePair.getLeft(), libraryNameValuePair.getRight());
+      if (optionalLibrary.isPresent()) {
+        Library library = optionalLibrary.get();
+        libraries.add(library);
```
Comment (adongare) [inline src/main/java/gov/cms/madie/madiefhirservice/services/MeasureBundleService.java:78]: Still thinking about checking duplicates. not sure if there will be duplicates at this point.

File: src/main/java/gov/cms/madie/madiefhirservice/services/LibraryService.java [hunk 2]
```diff
@@ -99,4 +104,26 @@ public Library createLibraryResourceForCqlLibrary(CqlLibrary cqlLibrary) {
     hapiFhirServer.createResource(library);
     return library;
   }
+
+  public void getIncludedLibraries(String cql, List<Library> libraries) {
+    if (StringUtils.isBlank(cql) || libraries == null) {
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/madiefhirservice/services/LibraryService.java]: What's the value of requiring the resulting ArrayList vs returning a fresh ArrayList?

Comment (jkotanchik-SB) [reply src/main/java/gov/cms/madie/madiefhirservice/services/LibraryService.java]: Though unlikely, I see a possible problem when the caller passes in a null value for `libraries`. The method will cleanly exit and give no indication to the caller that the result was not populated.

Instead, suggest throwing `IllegalArguementException` when `libraries` is null, or refactoring to return the ArrayList and initialize within the method.

Comment (adongare) [reply src/main/java/gov/cms/madie/madiefhirservice/services/LibraryService.java]: good point, changed this. Thanks!

**Review by jkotanchik-SB — APPROVED** [review]
